hazelcast-client/src/test/java/com/hazelcast/client/spi/ProxyFactoryTest.java

```diff
@@ -28,100 +30,185 @@
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
 import org.junit.After;
-import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
-import static org.junit.Assert.fail;
+import static com.hazelcast.client.impl.ClientTestUtil.getHazelcastClientInstanceImpl;
+import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
 public class ProxyFactoryTest {
```

Comment:
The test creates proxies with and without `ClientContext` to ensure both ways are working.